File: Client/src/app/shared/_models/basket.ts
import { v4 as uuid } from 'uuid';

export class Basket {
  id = uuid();
  items: BasketItem[] = [];
  clientSecret?: string;
  paymentIntentId?: string;
  deliveryMethod?: number;
  shippingPrice?: number;
}

export interface BasketItem {
  id: number;
  name: string;
  price: number;
  quantity: number;
  pictureUrl: string;
  brand: string;
  type: string;
}

export interface BasketTotals {
  shipping: number;
  subtotal: number;
  total: number;
}

File: Core/Entities/Basket.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace Core.Entities
{
    public class Basket
    {
        public string Id { get; set; }
        public List<BasketItem> Items { get; set; }
        public int? DeliveryMethod { get; set; }
        public string ClientSecret { get; set; }
        public string PaymentIntentId { get; set; }
        public decimal ShippingPrice { get; set; }

        public Basket()
        {
            Items = new List<BasketItem>();
        }

        public Basket(string id)
            : this()
        {
            Id = id ?? throw new ArgumentNullException(nameof(id));
        }
    }
}
